src/bins.rs

```diff
@@ -43,79 +46,68 @@ impl BinFile {
             data.bin_path.join(&source_file_path)
         };
 
-        // Destination path is the install dir + base-name-version{.extension}
-        let dest_file_path = ctx.render("{ bin }-v{ version }{ binary-ext }")?;
-        let dest = data.install_path.join(dest_file_path);
+        // Versioned path is the install dir + base-name-version{.extension}
+        let verd_file_path = ctx.render("{ bin }-v{ version }{ binary-ext }")?;
+        let versioned = data.install_path.join(verd_file_path);
 
-        // Link at install dir + base-name{.extension}
-        let link = data
+        // Main file at install dir + base-name{.extension}
+        let main = data
             .install_path
             .join(&ctx.render("{ bin }{ binary-ext }")?);
 
         Ok(Self {
             base_name,
             source,
-            dest,
-            link,
+            versioned,
+            main,
         })
     }
 
-    pub fn preview_bin(&self) -> String {
+    pub fn main_filename(&self) -> String {
+        self.main.file_name().unwrap().to_string_lossy().into()
+    }
+
+    pub fn versioned_filename(&self) -> String {
+        self.versioned.file_name().unwrap().to_string_lossy().into()
+    }
+
+    pub fn versioned_preview(&self) -> String {
         format!(
             "{} ({} -> {})",
             self.base_name,
             self.source.file_name().unwrap().to_string_lossy(),
-            self.dest.display()
+            self.versioned.display()
         )
     }
 
-    pub fn preview_link(&self) -> String {
+    pub fn main_preview(&self) -> String {
         format!(
             "{} ({} -> {})",
             self.base_name,
-            self.link.display(),
-            self.link_dest().display()
+            self.source.file_name().unwrap().to_string_lossy(),
+            self.main.display(),
         )
     }
 
-    pub fn install_bin(&self) -> Result<(), BinstallError> {
-        // TODO: check if file already exists
-        debug!(
-            "Atomically install file from '{}' to '{}'",
-            self.source.display(),
-            self.dest.display()
-        );
-        atomic_install(&self.source, &self.dest)?;
+    pub fn install_only_main(&self) -> Result<(), BinstallError> {
+        info!("Install: {}", self.main_preview());
+        try_remove(&self.main);
+        install_copy(&self.source, &self.main)?;
```

Comment:
Honestly, I prefer atomic `mv` to using reflink, though the later is also fine I guess.

Is there any particular reason why we prefer to use reflink here?